### PR TITLE
Add Headlamp integration post to blog

### PR DESCRIPTION
**Pull request by joaquimrocha**
This is a proposal for a repost of the blog post https://headlamp.dev/blog/2024/08/08/opencost-plugin .
I wasn't sure about who to add as an author, so I added the OpenCost team. Please let me know if you need some changes to give a better idea that the post is a repost from Headlamp's context.

### CODE REVIEW

**Comment by joaquimrocha** [comment]
cc/ @yolossn

File: blog/2024-08-14-headlamp/index.mdx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+      namespaceName: <Prometheus-Namespace>
+      port: 9090
+      serviceName: <Prometheus-Service>
+```
+
+:::note
+Replace `<Prometheus-Namespace>` and `<Prometheus-Service>` with the namespace in which you have installed prometheus and prometheus service name.
+:::
+
+## Installing KubeCost plugin from the plugin Catalog
```
Comment (KC-Jenn) [inline blog/2024-08-14-headlamp/index.mdx:58]: Would you please update Kubecost to have a lowercase c?

File: blog/2024-08-14-headlamp/index.mdx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+![Screenshot showing OpenCost chart in Headlamp](./img/opencost-chart-in-headlamp.png)
+
+## Configuring how data is displayed
+
+OpenCost's settings (under Settings > Plugins > OpenCost) provide options to adjust how the data is shown in the UI, ie. what currency symbol to be used, or timespan for which the cost has to be displayed. This way, users can easily adapt how OpenCost is displayed for them.
+
+## Conclusion
+
+We believe this plugin is useful for many users who want to keep their costs monitored close to their Kubernetes resources. We are looking forward to receiving feedback and contributions on how we can make the UI even more useful with OpenCost and other CNCF tools, so reach out to us and get involved in Headlamp!
```
Comment (KC-Jenn) [inline blog/2024-08-14-headlamp/index.mdx:86]: Would you please add how you would like users to reach out to you? E.g. a URL or email address?

**Review by KC-Jenn — CHANGES_REQUESTED** [review]
Just a couple of small changes, please!

File: blog/2024-08-14-headlamp/index.mdx [hunk 3]
```diff
@@ -0,0 +1,86 @@
+---
+slug: headlamp
+title: "Repost: Streamlining Kubernetes Cost Management with the New OpenCost Plugin for Headlamp"
+authors: [team]
+tags: [opencost, headlamp, kubernetes, ui, integrations]
+---
+<head>
+    <link rel="canonical" href="https://headlamp.dev/blog/2024/08/08/opencost-plugin"/>
```
Comment (dwbrown2) [inline blog/2024-08-14-headlamp/index.mdx:8]: Anything else we need to do to properly support a repost? I'm less familiar here.

File: blog/2024-08-14-headlamp/index.mdx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+Reposted from the [Headlamp blog](https://headlamp.dev/blog/) post: [Streamlining Kubernetes Cost Management with the New OpenCost Plugin for Headlamp](https://headlamp.dev/blog/2024/08/08/opencost-plugin), by Santhosh Nagaraj
+
+![OpenCost + Headlamp](./img/splash.jpg)
+
+[OpenCost](https://www.opencost.io/) is one of CNCF's leading open-source projects that offers real-time native cost monitoring for Kubernetes environments, providing visibility to developers and companies on the costs linked to their cloud-native applications. With OpenCost, one can trace the costs of single workloads, namespaces, and even particular labels in your Kubernetes cluster. The continuous granular cost visibility that OpenCost provides help in effectual usage of resources and better budgeting.
+
+<!--truncate-->
+
+Given how useful OpenCost is, we created a [Headlamp](https://headlamp.dev/) plugin for OpenCost to help Headlamp users track their cluster resource costs. This integration aims to solve the visibility problem between cost and cluster management, enabling developers to make more informed decisions.
```
Comment (dwbrown2) [inline blog/2024-08-14-headlamp/index.mdx:19]: Would it be possible to reframe this from "we created a" to "the Headlamp team" or something along those lines?

File: blog/2024-08-14-headlamp/index.mdx [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+```
+
+Installing OpenCost:
+
+```yaml
+opencost:
+  prometheus:
+    internal:
+      enabled: true
+      namespaceName: <Prometheus-Namespace>
```
Comment (dwbrown2) [inline blog/2024-08-14-headlamp/index.mdx:49]: It could be nice to have the defaults "just work" here?

**Review by dwbrown2 — COMMENTED** [review]
Awesome post! Just a handful of questions/comments from me.

File: blog/2024-08-14-headlamp/index.mdx (hunk 5 above)
Comment (joaquimrocha) [reply blog/2024-08-14-headlamp/index.mdx:49]: I think changing that would require some testing. If this makes sense I think we should leave it as is.

File: blog/2024-08-14-headlamp/index.mdx (hunk 4 above)
Comment (joaquimrocha) [reply blog/2024-08-14-headlamp/index.mdx:19]: Yeah. I wondered about this but thought other posts we using that person as well.
I can change the version mention to something along what you suggest.

File: blog/2024-08-14-headlamp/index.mdx (hunk 3 above)
Comment (joaquimrocha) [reply blog/2024-08-14-headlamp/index.mdx:8]: I am actually not sure. I just copied one of the reposts you had and changed accordingly.

File: blog/2024-08-14-headlamp/index.mdx (hunk 2 above)
Comment (joaquimrocha) [reply blog/2024-08-14-headlamp/index.mdx:86]: I will add something.

File: blog/2024-08-14-headlamp/index.mdx (hunk 5 above)
Comment (dwbrown2) [reply blog/2024-08-14-headlamp/index.mdx:49]: Yep, don't feel strongly here... just a potential future improvement

File: blog/2024-08-14-headlamp/index.mdx (hunk 1 above)
Comment (joaquimrocha) [reply blog/2024-08-14-headlamp/index.mdx:58]: I think this was a typo. Should be OpenCost.

**Comment by joaquimrocha** [comment]
Thanks @dwbrown2 / @KC-Jenn  . I have updated the post. Let me know if these changes address what you requested or you need something different.

**Comment by joe-terrible** [comment]
Hi @joaquimrocha - one final suggestion: you might want to add a couple of sentences at the top covering what Headlamp is and the benefit of using it. The audience will be familiar with OpenCost and Kubecost, but might not know much about the Headlamp app or your K8s UI.

**Comment by joaquimrocha** [comment]
@joe-terrible , makes sense. Done.

**Review by KC-Jenn — APPROVED** [review]
Thanks for making the updates!

**Review by dwbrown2 — APPROVED** [review]

File: blog/2024-08-14-headlamp/index.mdx (hunk 3 above)
Comment (dwbrown2) [reply blog/2024-08-14-headlamp/index.mdx:8]: I'm not an expert but looks fine...

Comment (dwbrown2) [reply blog/2024-08-14-headlamp/index.mdx:8]: cc @joe-terrible @dzeiskubecost

**Comment by joaquimrocha** [comment]
Updated the date for the post as agreed by email.

**Pull request merged**